control-plane-operator/controllers/hostedcontrolplane/v2/machine_approver/deployment.go

```diff
@@ -0,0 +1,30 @@
+package machineapprover
+
+import (
+	"fmt"
+
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/manifests"
+	component "github.com/openshift/hypershift/support/controlplane-component"
+	"github.com/openshift/hypershift/support/util"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	kubeconfigVolumeName = "kubeconfig"
+)
+
+func adaptDeployment(cpContext component.WorkloadContext, deployment *appsv1.Deployment) error {
+	hcp := cpContext.HCP
+
+	util.UpdateContainer(ComponentName, deployment.Spec.Template.Spec.Containers, func(c *corev1.Container) {
+		c.Args = append(c.Args, fmt.Sprintf("--machine-namespace=%s", hcp.Namespace))
+	})
+
+	util.UpdateVolume(kubeconfigVolumeName, deployment.Spec.Template.Spec.Volumes, func(v *corev1.Volume) {
+		v.Secret.SecretName = manifests.KASServiceKubeconfigSecret(hcp.Namespace).Name
+	})
```

Comment:
I don't think this is need once the kubeconfig volume is added to the manifest.

Reply:
dropped it. Like for sa kubeconfig, as a component I'd expect to have a way to express intent to self service a kubeconfig so I don't need to hand craft it manually and I don't need to figure there's a secret named service-network-admin-kubeconfig or if their permissions are a good fit. I'll follow up to solve this.

Reply:
makes sense